File: 程式設計：資料結構答案檔/Python/202.py
def inorder_tree(idx):
    if (idx < len(lst)) and lst[idx] != '':
        inorder_tree(2*idx)
        print(lst[idx], end='')
        inorder_tree(2*idx + 1)

def leaf_node():
    leaves = []
    
    for i in range(1, len(lst)):
        if (2*i >= len(lst)) or (lst[2*i] == ''):
            if (2*i+1 >= len(lst)) or (lst[2*i+1] == ''):
                leaves.append(lst[i])
    
    leaves.sort()
    print("".join(leaves))

inp = input()
lst = [None] + inp.split(',')

inorder_tree(1)

print("\n", end='')
leaf_node()

File: 程式設計：資料結構答案檔/C#/202.cs
using System;
using System.Collections.Generic;
using System.Linq;

class BinaryTreeTraversal {
    static string[] lst;

    static void InorderTree(int idx) {
        if (idx < lst.Length && lst[idx] != "") {
            InorderTree(2 * idx);
            Console.Write(lst[idx]);
            InorderTree(2 * idx + 1);
        }
    }

    static void LeafNode() {
        List<string> leaves = new List<string>();
        
        for (int i = 1; i < lst.Length; i++) {
            if ((2 * i >= lst.Length || lst[2 * i] == "") && (2 * i + 1 >= lst.Length || lst[2 * i + 1] == "")) {
                leaves.Add(lst[i]);
            }
        }
        
        leaves.Sort();
        Console.WriteLine(string.Join("", leaves));
    }

    static void Main(string[] args) {
        string inp = Console.ReadLine();
        lst = new string[] {null}.Concat(inp.Split(',')).ToArray();

        InorderTree(1);
        Console.WriteLine();
        LeafNode();
    }
}
